### DISCUSSION TITLE
General Question regarding Virtual Environments

### DISCUSSION THREAD

**Discussion by NewCoderrrrr** [smartcontractkit/full-blockchain-solidity-course-py#1272] [Q&A] [answered]
I am wondering if there is any way to set a virtual environment as like the default for all VSCode projects or maybe the demos directory or something along those lines? As of now I have to do it every single time I create a new folder which is inconvenient. I also have to reinstall a bunch of things once I set the interpreter to the (myvenv) at the start. I setup the venv using the command on macOS "python3 -m venv myvenv". 

I read this article: https://levelup.gitconnected.com/how-to-work-with-vs-code-and-virtual-environments-in-python-5aa4f65afb3d 

I only partially followed this article but I noticed that they mention the requirements.txt file and then they have a bunch of things in there. Is there anyway to acquire one of those for the things we do in this course so then we can download it every time? 
Thanks

**Comment by cromewar** [comment, marked as answer]
Hello @NewCoderrrrr 

Well the best way to do this is to run vscode from the terminal, instead of opening it using the contextual menu.
1. Activate your virutalenv.
2. run code .
3. vscode should now be setup using the virtualenv you have activated on the terminal.

Also take something ins consideration, you don't need a new virtualenv for each single project you are going to work on, from lesson 4 to 12 you can just use the same as there are not big changes on packages and dependencies.

To create a requirements.txt just use this command (after activate the env you want to export the requirements for):

`pip3 freeze > requirements.txt `

> Also another question: How do I leave the GitHub portion of VSCode? I did the "git add ." command and ever since then my entire 
 VSCode is like color coordinated.

I did not understand this question could you be more specific please?

**Reply by NewCoderrrrr** [reply]
@cromewar Thank you I will try this, as for the color coordinated thing I was mentioned, if you go to 6:05:40 in the video and watch for like a minute you will see what I mean. Patricks whole VSCode became color coordinated like all the files etc. Is there anyway to get rid of this?

**Reply by NewCoderrrrr** [reply]
I figured it out. You have to type: rm -rf .git

**Reply by NewCoderrrrr** [reply]
Ok so I only just got around to actually trying this. I installed the code command path. Now I just open the terminal on my Mac or is it in VSCode and then type code . and all of my code is now in a venv? @cromewar The code path said it downloaded to "/usr/local/bin/code" does this mean I have to cd .. and cd into this as in usr -> local -> bin -> code, because the code . command wasn't working. I just find that these venv's are really confusing and I don't understand the benefits.

**Reply by cromewar** [reply]
Follow this guide @NewCoderrrrr https://stackoverflow.com/questions/30065227/run-open-vscode-from-mac-terminal

**Reply by NewCoderrrrr** [reply]
Ok I got the code . command to work with one of the recommendations in the stackoverflow you linked. Now i'm wondering how to "activate your virtual environment" through the terminal on MacOS.  
Do I do something like: 
```
1.   # Creates the virtual environment.
	virtualenv venv
  
2.  # Activates virtual environment.
	source venv/bin/activate

```
Is this something that I have to do every time before typing code . 

Another thing is that after running code ., it warned me that the file is corrupted. I got the code . command to work by changing the bash_file to include 

code () { VSCODE_CWD="$PWD" open -n -b "com.microsoft.VSCode" --args $* ;} 

Was this safe?

**Reply by cromewar** [reply]
to activate the environment you use:
`source envFolder/bin/activate`
adding to path the vscode command is normal an totally safe.

**Reply by n4n0b1t3** [reply]
you can setup VSC top automatically activate the virtual env when in your project. Have a look at the end of the before mentioned article https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/1109

**Comment by n4n0b1t3** [comment]
This is the way to go: https://github.com/smartcontractkit/full-blockchain-solidity-course-py/discussions/1109
just updated it, you just need one virtual environment for one setup and can run many different apps with it.

**Reply by NewCoderrrrr** [reply]
Thanks I will read this  @n4n0b1t3

**Reply by NewCoderrrrr** [reply]
IDK how to follow this on MacOS

**Reply by n4n0b1t3** [reply]
I think the only difference should be the  Visual C++ build tools everything else should be much the same. At what point didn't it work on your machine?
